pkg/deployer/baremetal/artifacts.go

```diff
@@ -176,7 +182,7 @@ func (am *ArtifactManager) installGreptime(artifactFile, binDir string) error {
 }
 
 func (am *ArtifactManager) download(typ ArtifactType, version, pkgDir string) (string, error) {
-	downloadURL, err := am.artifactURL(typ, version)
+	downloadURL, err := am.artifactURL(typ, version, ZipExtension)
```

Comment:
Should we always use `*.zip`? It seems that the upstream release the packages base on different arch and os:

- `*.darwin-amd64.zip`
- `*.darwin-arm64.zip`
- `*.linux-amd64.tar.gz`
- `*.linux-arm64.tar.gz`
- `*.windows.amd64.zip`(but the greptimedb don't support windows now);

Based on the above format, I think we should use `*.zip` on darwin, `*.tar.gz` on linux.

Reply:
I am not sure if other installation packages need to be added to the project in the future. If the installation packages have already been collected, you can organize their patterns to determine the list of required suffix names. Alternatively, before entering its method, determine the suffix name based on `OS` and `Arch`, and then pass it to the function to avoid two subsequent HTTP requests.

Reply:
@sjcsjc123 I think we can determine the ext suffix base on the os and arch.